app/server/me_api/misc.py

```diff
@@ -282,4 +286,106 @@ def post(self):
             'data': create_payout_account_response
         }
 
-        return make_response(jsonify(response_object)), 201
\ No newline at end of file
+        return make_response(jsonify(response_object)), 201
+
+
+class PoliPaymentsAPI(MethodView):
+    @requires_auth
+    def get(self, reference):
+        reference = request.args.get('reference')
+        if reference:
+
+            fiat_ramp = FiatRamp.query.filter_by(payment_reference=reference).first()
+            if fiat_ramp is None:
+                return make_response(jsonify({'message': 'Could not find payment for reference {}'.format(reference)})), 400
+
+            poli_link = fiat_ramp.payment_metadata['poli_link']
+            try:
+                get_poli_link_status_response = get_poli_link_status(poli_link=poli_link.split('/')[-1])
+
+            except PoliPaymentsError as e:
+                response_object = {
+                    'message': str(e)
+                }
+                return make_response(jsonify(response_object)), 400
+
+            status = get_poli_link_status_response['status']
+
+            if fiat_ramp.payment_status == FiatRampStatusEnum.PENDING:
+                if status == 'Completed':
+                    # "The full amount has been paid"
+                    individual_recipient_user = User.query.execution_options(show_all=True).get(fiat_ramp.authorising_user_id)
+                    deposit = make_deposit_transfer(
+                        transfer_amount=fiat_ramp.payment_amount,
+                        token=fiat_ramp.token,
+                        receive_account=individual_recipient_user,
```

Comment:
`receive_user` is probably less ambiguous - aware that this one is a hangover from my bad naming lol!